examples/test_util/ESmry.cpp

```diff
@@ -0,0 +1,436 @@
+        arrayInd.push_back(tmpVect);
+    }
+    
+    int n = nFiles - 1;
+    
+    while (n >= 0){
+
+        auto smry = smryArray[n];
+
+        EclFile smspec(std::get<0>(smry));
+        smspec.loadData();
+
+        std::vector<int> dimens = smspec.get<int>("DIMENS");
+
+        nI = dimens[1];
+        nJ = dimens[2];
+        nK = dimens[3];
+
+        std::vector<std::string> keywords = smspec.get<std::string>("KEYWORDS");
+        std::vector<std::string> wgnames = smspec.get<std::string>("WGNAMES");
+        std::vector<int> nums = smspec.get<int>("NUMS");
+
+        std::set<std::string>::iterator it;
+
+        for (size_t i=0; i < keywords.size(); i++) {
+            std::string keyw = makeKeyString(keywords[i], wgnames[i], nums[i]);
+            it = std::find(keywList.begin(), keywList.end(), keyw);
+
+            if (it != keywList.end()){
+                arrayInd[n][i] = distance(keywList.begin(), it);
+            }
+        }
+        
+        n--;
+    }
+
+    // param array used to stor data for the object, defined in the private section of the class 
+    param.assign(keywList.size(), {});
+    
+    int fromReportStepNumber = 0;
+    int reportStepNumber = 0;
+    int toReportStepNumber;
+
+    float time = 0.0;
+    int step = 0;
+
+    n = nFiles - 1;
+
+    while (n >= 0){
+
+        reportStepNumber = fromReportStepNumber;
+
+        if (n > 0) {
+            auto rstFrom = smryArray[n-1];
+            toReportStepNumber = std::get<1>(rstFrom);
+        } else {
+            toReportStepNumber = std::numeric_limits<int>::max();
+        }
+
+        std::string smspecFile = std::get<0>(smryArray[n]);
+        std::string unsmryFile = smspecFile.substr(0, smspecFile.size() - 6) + "UNSMRY";
+
+        EclFile unsmry(unsmryFile);
+        unsmry.loadData();
+
+        std::vector<EclFile::EclEntry> list1 = unsmry.getList();
+
+        // 2 or 3 arrays pr time step.
+        //   If timestep is a report step:  MINISTEP, PARAMS and SEQHDR
+        //   else : MINISTEP and PARAMS
+
+        // if summary file starts with a SEQHDR, this will be ignored
+
+        int i = 0;
+
+        if (std::get<0>(list1[0]) == "SEQHDR") {
+            i = 1;
+        }
+
+        while  (i < static_cast<int>(list1.size())){
+
+            if (std::get<0>(list1[i]) != "MINISTEP"){
+                std::string message="Reading summary file, expecting keyword MINISTEP, found '" + std::get<0>(list1[i]) + "'";
+                throw std::invalid_argument(message);
+            }
+
+            std::vector<int> ministep = unsmry.get<int>(i);
+
+            i++;
+
+            if (std::get<0>(list1[i]) != "PARAMS") {
+                std::string message="Reading summary file, expecting keyword PARAMS, found '" + std::get<0>(list1[i]) + "'";
+                throw std::invalid_argument(message);
+            }
+
+            std::vector<float> tmpData = unsmry.get<float>(i);
+
+            time = tmpData[0];
```

Comment:
You should check the units for the time - the value you have stored in the `time` variable can be in either seconds or days (I have a vague feeling you can even get it in years!).

Reply:
i don't understand what you mean here.

Reply:
Well - unless I am mistaken, the quantity you fetch here and store in variable `time` is the duration of the simulation - in "reservoir time", that can be expressed in different units - i.e. `DAYS` or `SECONDS` - and if you want express results using real world time points like `21/02/2017` you need to know which units of time are used.

But of course - you could say I don't care - it is just a float?

Reply:
The vectors   
    std::vector<int> seqIndex;
    std::vector<float> seqTime;

Are used to store index and time at reports steps (ignoring ministeps). These are not used in the current version of the class, but the plan is to add a public member function 

  const std::vector<float>& get_reports_steps_only(const std::string& name) const;

or something like this. This will return a vectors sampled at results at report steps only. 

This will be useful when comparing simulations with different time stepping.